x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engines/engines_list_logic.ts

```diff
@@ -145,10 +156,16 @@ export const EnginesListLogic = kea<MakeLogicType<EngineListValues, EnginesListA
       (status: EngineListValues['deleteStatus']) => [Status.LOADING].includes(status),
     ],
     isLoading: [
-      () => [selectors.status],
-      (status: EngineListValues['status']) => [Status.LOADING, Status.IDLE].includes(status),
+      () => [selectors.status, selectors.isFirstRequest],
+      (status: EngineListValues['status'], isFirstRequest: EngineListValues['isFirstRequest']) =>
+        [Status.LOADING, Status.IDLE].includes(status) && isFirstRequest,
     ],
     results: [() => [selectors.data], (data) => data?.results ?? []],
+    hasNoEngines: [
+      () => [selectors.data, selectors.results],
+      (data: EngineListValues['data'], results: EngineListValues['results']) =>
+        (data?.isInitialRequest && results.length === 0) ?? false,
+    ],
```

Comment:
Alternative to my `isFirstRequest` suggestion above I think we could just change this selector to:

```suggestion
    hasNoEngines: [
      () => [selectors.results, selectors.parameters],
      (results: EngineListValues['results'], parameters: EngineListValues['parameters']) =>
        results.length === 0 && parameters.meta.from === 0 && !parameters.searchQuery,
    ],
```

To determine if we should show no engines vs the table with no results. We may want to add an `isLoading` check there, but I'd have to see what we render when `isLoading === true`

Reply:
If this results in flashing the empty engines component after you clear a search query, then it would be better to use the `isFirstRequest` above 
```suggestion
    hasNoEngines: [
      () => [selectors.isFirstRequest, selectors.results],
      (isFirstRequest: EngineListValues['isFirstRequest'], results: EngineListValues['results']) =>
        isFirstRequest && results.length === 0,
    ],
```

Reply:
I think `hasNoEngines` would always be `false` after the first request. This might cause the page not to redirect when all the engines are deleted after the first request

Reply:
> Alternative to my `isFirstRequest` suggestion above I think we could just change this selector to:
> 
> To determine if we should show no engines vs the table with no results. We may want to add an `isLoading` check there, but I'd have to see what we render when `isLoading === true`

From debug logs I see that `isLoading` is `true` on initial page load and `isLoading` is changed to `false` when  request is made to fetchEngines. This could be due the change added in 
https://github.com/elastic/kibana/blob/d6d6c200f6850d34edf2aeceb960e31b9141dd0d/x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engines/engines_list_logic.ts#L140

Removing the `isFirstRequest` check from the above line would cause list page to reload on every search query.